kubernetes/kubernetes_service.go

```diff
@@ -65,6 +67,19 @@ func (in *IstioClient) GetNamespaces() (*v1.NamespaceList, error) {
 	return namespaces, nil
 }
 
+func (in *IstioClient) GetProjects() (*osv1.ProjectList, error) {
+
+	result := &osv1.ProjectList{}
+
+	err := in.k8s.RESTClient().Get().Prefix("apis", "project.openshift.io", "v1", "projects").Do().Into(result)
```

Comment:
This is plain/raw REST.
I suggest to add this API into the os_route_client.go as it is also used to queried OpenShift specific API, just for consistency.
Perhaps os_route_client.go is more generic and can be refactored to os_client.go.

Reply:
Yeah, that makes sense, the os_route_client.go is a bit messy as it deals with plain json and casting things to interfaces.